### ISSUE TITLE
DX: css prop similar to emotion, styled-components

### ISSUE THREAD

**Issue by kevinSuttle** [cssinjs/jss#1092]
__Is your feature request related to a problem? Please describe.__
No. 

__Describe the solution you'd like__
With [emotion](https://emotion.sh/docs/css-prop) and [styled-components](https://www.styled-components.com/docs/api#css-prop), you can provide a `css` propthat allows you to override a given component's styles. 

Both libraries require a babel-plugin, and emotion requires a [`jsx` pragma.](https://emotion.sh/docs/css-prop#jsx-pragma) 

__Are you willing to implement it?__
Willing. Yes. Able. _Mehhhhh_

**Comment by kevinSuttle** [comment]
I wonder if a [`babel-macro`](https://github.com/emotion-js/emotion/tree/master/packages/babel-plugin-emotion#babel-macros) would be an ideal tool for this job 🤔

**Comment by jeremy-coleman** [comment]
This already exists in react jss , the prop is just called classes ..

**Comment by kevinSuttle** [comment]
If that's the case, then 🎉

**Comment by kof** [comment]
@jeremy-coleman not exactly, the difference is classes is a user level api that user needs to use to pass a className to react built-in component, where prop basically gets renamed to a className and it does the `css()` function call. This isn't a big feature in my eyes, it's more or less a syntactic sugar.

Another point is that emotion/SC accepts a style object right in the render, which seems convenient from the user perspective, but on the other hand brings 2 disadvantages:
1. Object is created during render function, so a reference is new and compliled styles can't be matched even if it's not changed by reference, so to hit the cache one needs to stringify that object every time and compare the string. We pay here every time the stringification cost on every render.
2. While creating a small styled component with inlined styles isn't a problem, creating a bigger one and using a lot of it will create a cognitive overhead, because jsx and css will be a wild mix. In that case I would prefer styles being outside of render function to see html structure without styles and styles somewhere before/after the component, but since this is not enforced by the API a larger team will have both in different ways and it will inevitably lead to less readable jsx over time.

**Comment by hendrul** [comment]
css prop is not the only one out there, I would like to use [twin.macro's](https://github.com/ben-rogerson/twin.macro) tw prop for example, to be able to style using tailwind. The point is, there is a need to use this kind of extensions.
